src/createEffectCapableStore.js

```diff
@@ -68,13 +13,67 @@ export const wrapReplaceReducer = store => nextReducer =>
  * @returns {Function} Store factory
  */
 export default createStore => (rootReducer, initialState) => {
-  const store = createStore(enhanceReducer(rootReducer), new AppStateWithEffects(initialState, []));
+  // Reference to the store dispatch function.
+  let dispatch;
```

Comment:
If it was a production code: We would need to get rid of this mutable reference, maybe some thunk could help